client/components/happychat/test/composer.jsx

```diff
@@ -22,61 +25,74 @@ describe( '<Composer />', () => {
 					translate={ noop }
 				/>
 			);
-			wrapper.find( 'textarea' ).simulate( 'change', { target: { value: 'hey' } } );
+			const el = screen.getByRole( 'textbox', { name: /support request/i } );
+			await user.type( el, 'hey' );
 			expect( onSetCurrentMessage ).toHaveBeenCalled();
 			expect( onSendTyping ).toHaveBeenCalled();
 			expect( onSendNotTyping ).not.toHaveBeenCalled();
 		} );
 
-		test( 'should call onSetCurrentMessage property and send a noTyping event if message is empty', () => {
+		test( 'should call onSetCurrentMessage property and send a noTyping event if message is empty', async () => {
+			const user = userEvent.setup();
 			const onSendNotTyping = jest.fn();
 			const onSendTyping = jest.fn();
 			const onSetCurrentMessage = jest.fn();
-			const wrapper = mount(
-				<Composer
-					message={ '' }
+			const WrappedWithMessageState = ( props ) => {
+				const [ message, setMessage ] = useState( 'x' );
+				props.onSetCurrentMessage.mockImplementation( ( value ) => setMessage( value ) );
+				return <Composer { ...props } message={ message } />;
+			};
+			render(
+				<WrappedWithMessageState
 					onSetCurrentMessage={ onSetCurrentMessage }
 					onSendTyping={ onSendTyping }
 					onSendNotTyping={ onSendNotTyping }
 					translate={ noop }
 				/>
 			);
-			wrapper.find( 'textarea' ).simulate( 'change', { target: { value: '' } } );
+			const el = screen.getByRole( 'textbox', { name: /support request/i } );
+			await user.click( el );
+			await user.keyboard( '{Backspace}' );
 			expect( onSetCurrentMessage ).toHaveBeenCalled();
 			expect( onSendTyping ).not.toHaveBeenCalled();
 			expect( onSendNotTyping ).toHaveBeenCalled();
 		} );
 	} );
 
 	describe( 'onKeyDown event', () => {
-		test( 'should call message and noTyping props if message is not empty', () => {
+		test( 'should call message and noTyping props if message is not empty', async () => {
+			const user = userEvent.setup();
 			const onSendMessage = jest.fn();
 			const onSendNotTyping = jest.fn();
-			const wrapper = mount(
+			render(
 				<Composer
 					message={ 'hey' }
 					onSendMessage={ onSendMessage }
 					onSendNotTyping={ onSendNotTyping }
 					translate={ noop }
 				/>
 			);
-			wrapper.find( 'textarea' ).simulate( 'keydown', { which: 13, preventDefault: () => {} } );
```

Comment:
This test used to press `Enter`, and now we click the submit button instead. Should we maintain the original behavior?

Reply:
That's true. Hitting Enter key on a `textarea` doesn't do what we're intending to test here though (submitting the form). That's the reason I opted for pressing the button. Assertions are still the same. Alternatively we could simulate a `CMD + Enter` to get the same behaviour.

Reply:
Hm, I disagree here. Usually, pressing `Enter` inside a `textarea` doesn't trigger a submit, but this is a special case. This textarea isn't intended to have newlines in it; rather when the user presses `Enter`, the message should be sent. This is indicated by the `keyDown` event handler on the textarea: 

https://github.com/Automattic/wp-calypso/blob/f2e6f88ae65c8571bbac21b284c29b48ec987598/client/components/happychat/composer.jsx#L45

So, I'd recommend keeping the same behavior.

Assuming you might have tried `await user.keyboard( '[Enter]' );` or `await user.type( textarea, '[Enter]' );` and it didn't work, this seems to be a bug with `@testing-library/user-event` and `jsdom` in particular. I haven't researched too much, but had a similar problem and solved it by falling back to `fireEvent()` for this particular scenario. See https://github.com/WordPress/gutenberg/pull/43147/files#diff-f133ef8bc484506e19d14be3b9f281b93b657b7fc472b757569e9b3548329483R168

Reply:
You're absolutely right 🤦🏻 I missed that detail. I'll adjust the PR accordingly 👍🏻

Reply:
@tyxla FYI the reason for why [`event.keyCode`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode) and [`event.which`](https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/which) (used here) don't work with RTL is (probably) because both attributes are deprecated/non-standard and it's recommended to use `.key` or `.code`.

Reply:
Added https://github.com/Automattic/wp-calypso/pull/66599/commits/65ca879ab6b57ead171bc54c76b4e6b72d03b2ac, let me know what you think :)